src/components/ShareTrajectoryModal/index.tsx

```diff
@@ -29,27 +35,27 @@ const ShareTrajectoryModal = ({
     timeUnits,
     displayTimes,
 }: ShareTrajectoryModalProps): JSX.Element => {
-    /** URL Link Section */
-    const roundToTimestepPrecision = (
-        input: number,
-        timestep: number
-    ): number => {
-        const precision = (timestep.toString().split(".")[1] || "").length;
-        const multiplier = Math.pow(10, precision);
-        return Math.round(input * multiplier) / multiplier;
-    };
+    enum TimeToUse {
+        BEGINNING = "beginning",
+        USER_ENTERED = "user-entered",
+    }
+    const [startTimeToUse, setStartTimeToUse] = React.useState(
+        TimeToUse.BEGINNING
+    );
+    const currentTime = roundToTimeStepPrecision(
+        displayTimes.roundedTime,
+        displayTimes.roundedTimeStep
+    );
 
-    const handleAllowTimeInput = (): void => {
-        const timeValue = allowTimeInput ? lastEnteredNumber : currentTime;
-        setLinkUrl(
-            editUrlParams(
-                window.location.href,
-                timeValue.toString(),
-                URL_PARAM_KEY_TIME
-            )
-        );
-        setAllowTimeInput((allowTimeInput) => !allowTimeInput);
-    };
+    const [userEnteredTime, setUserEnteredTime] = React.useState(currentTime);
+
+    const startTime =
+        startTimeToUse === TimeToUse.BEGINNING ? 0 : userEnteredTime;
+    const linkUrl = editUrlParams(
+        window.location.href,
+        startTime.toString(),
+        URL_PARAM_KEY_TIME
+    );
```

Comment:
this is completely derived from state, so you don't need to also set it in state